core/src/types/sign.rs

```diff
@@ -0,0 +1,95 @@
+//! Types for signing
+
+use std::cmp::Ordering;
+
+use data_encoding::HEXUPPER;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use super::address::Address;
+use super::key::common;
```

Comment:
It might make sense to `pub use` here. If I were a new dev and saw this file, I would assume I could fine keys and signature types in here.